average.js

```diff
@@ -1,6 +1,20 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  //return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+
+
+  var total = 0;// variable to store total sum of array:numbers
+  var count = 0;// total elements in array:numbers
+  var n = numbers.length;//length of array:numbers
+  for(let i=0;i<n;++i) {
+      if(isNaN(numbers[i])) continue;//check to skip NaN: not a Number
+      total += numbers[i];
+      count++;
+  }
+
+  //rounded to 2 digits after decimal
+  var average = ((total / count).toFixed(1));
+  return parseFloat(average); 
 }
 
 module.exports = {average};
```

Comment:
Try a new test: The tests in `test/average.test.js` express the functionality of the `average` function.
Consider this situation: IoT sensors sometimes report a lot of variation in their readings. If you consider that the sensor is reading out battery temperature, there cannot be sudden variation in reality. Hence, a lot of variation in the readings would mean a defective sensor.
Yet, our average function would compute the average as though everything was ok.
Add functionality: Come up with a way to return NaN from the average function, if the variation is too much.
You will need to:

- determine the meaning of 'too much'. E.g., the difference between max and min, or the difference in successive readings. Pick the method you feel is best. Include any references you used for it.
- implement a test to check that the average function returns NaN on high variance. Name the test appropriately.
- implement the functionality to pass the test.

Reply:
I tried to implement above case i.e. too much variation , but it failed showing code has complexity 6, allowed 3.
After searching I found out about Cyclomatic Complexity, can you please help me understand why my code has complexity of 6.
Attached is the screenshot.
Thank you
![complexity_error](https://user-images.githubusercontent.com/56825866/146340903-b41b0f36-49a4-4f88-8172-51544602da61.png)